File: backend/app/schemas/notification.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel


class NotificationCreate(BaseModel):
    user_id: int
    notification_type: str
    title: str
    message: str
    link: Optional[str] = None
    related_type: Optional[str] = None
    related_id: Optional[int] = None


class NotificationOut(BaseModel):
    id: int
    user_id: int
    notification_type: str
    title: str
    message: str
    link: Optional[str]
    related_type: Optional[str]
    related_id: Optional[int]
    is_read: bool
    read_at: Optional[datetime]
    created_at: datetime

    class Config:
        from_attributes = True


class NotificationCountOut(BaseModel):
    count: int


class NotificationListOut(BaseModel):
    notifications: List[NotificationOut]
    total: int
    unread_count: int

File: frontend/src/types/notification.ts
export interface Notification {
  id: number;
  user_id: number;
  notification_type: string;
  title: string;
  message: string;
  link: string | null;
  is_read: boolean;
  read_at: string | null;
  created_at: string;
  // Optional fields
  related_type?: string | null;
  related_id?: number | null;
}

export interface NotificationCount {
  count: number;
}

export interface NotificationSettings {
  email_enabled: boolean;
  web_push_enabled: boolean;
  reminder_24h: boolean;
  reminder_1h: boolean;
  notify_new_post: boolean;
  notify_comment_reply: boolean;
  notify_event_reminder: boolean;
  notify_event_update: boolean;
  notify_mention: boolean;
  notify_system: boolean;
}
